File: public/examples/integrations/mcp-servers/zoom_meetings_api/list_zoom_upcoming_events_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "ZoomMeetingsApi.ListZoomUpcomingEvents";

// Start the authorization process
const authResponse = await client.tools.authorize({tool_name: TOOL_NAME});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "start_date": "2025-10-01",
  "end_date": "2025-10-31",
  "records_per_page": 50,
  "event_type_to_query": "all",
  "pagination_token": "eyJwb2siOiJwYWdlMSIsImV4cCI6IjE2ODc5In0",
  "group_identifier": "grp_8a7f3c"
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));

File: public/examples/integrations/mcp-servers/zoom_meetings_api/list_zoom_upcoming_events_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "ZoomMeetingsApi.ListZoomUpcomingEvents"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=USER_ID,
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'start_date': '2025-10-01',
    'end_date': '2025-10-31',
    'records_per_page': 50,
    'event_type_to_query': 'all',
    'pagination_token': 'eyJwb2siOiJwYWdlMSIsImV4cCI6IjE2ODc5In0',
    'group_identifier': 'grp_8a7f3c'
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))
